services/agents/registry.go

```diff
@@ -456,6 +459,65 @@ func (r *Registry) SendSetStateRequest(ctx context.Context, pmmAgentID string) {
 	l.Infof("SetState response: %+v.", resp)
 }
 
+// CheckConnectionToService sends request to pmm-agent to check connection to service.
+func (r *Registry) CheckConnectionToService(ctx context.Context, service *models.Service, agent *models.Agent) error {
+	// TODO: extract to a separate struct to keep Single Responsibility principles.
+	l := logger.Get(ctx)
+	start := time.Now()
+	defer func() {
+		if dur := time.Since(start); dur > time.Second {
+			l.Warnf("CheckConnectionToService took %s.", dur)
+		}
+	}()
+
+	pmmAgentID := pointer.GetString(agent.PMMAgentID)
+	pmmAgent, err := r.get(pmmAgentID)
```

Comment:
Just `*agent.PMMAgentID`. It is ok panic and explode if caller passed nil agent, or agent without pmm-agent ID.

Reply:
I think returning error with agent looks better here than just agent. It's not obvious that pmmAgent might be a nil.